jobs/spFilteringJob.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"context"
+	"log"
+	"os"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/data-preservation-programs/spade-tenant/config"
+	"github.com/data-preservation-programs/spade-tenant/db"
+	"github.com/joho/godotenv"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"gorm.io/gorm"
+)
+
+type SP struct {
+	Version        int       `bson:"__v"`
+	Provider       string    `bson:"_id"`
+	AgentCity      string    `bson:"agentCity"`
+	AgentCountry   string    `bson:"agentCountry"`
+	AgentLatitude  float64   `bson:"agentLatitude"`
+	AgentLongitude float64   `bson:"agentLongitude"`
+	AgentRegion    string    `bson:"agentRegion"`
+	Date           time.Time `bson:"date"`
+	LatencyMs      float64   `bson:"latencyMs"`
+	Multiaddr      string    `bson:"multiaddr"`
+	TestId         string    `bson:"testId"`
+}
+
+func main() {
+
+	sps := setUpEnvAndFetchSpDataFromMongo()
```

Comment:
nit: move `godoteenv.Load()` to the beginning of `main`, then rename this function to simply `fetchSpData()`